src/sage/matroids/oriented_matroids/signed_subset_element.py

```diff
@@ -0,0 +1,850 @@
+            sage: E = M.elements()[0]; E
+            +: 0
+            -: 2
+            0: 1
+            sage: -E
+            +: 2
+            -: 0
+            0: 1
+        """
+        N = copy.copy(self)
+        N._p = self._n
+        N._n = self._p
+        return N
+
+    def __eq__(self, other):
+        """
+        Return whether two elements are equal.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E1 = M.elements()[0]
+            sage: E2 = M.elements()[1]
+            sage: E1 == E2
+            False
+            sage: E1 == E1
+            True
+        """
+        if isinstance(other, SignedSubsetElement):
+            if self._p == other._p \
+                    and self._n == other._n \
+                    and self._z == other._z:
+                return True
+        return False
+
+    def __ne__(self, other):
+        """
+        Return whether two elements are not equal.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E1 = M.elements()[0]
+            sage: E2 = M.elements()[1]
+            sage: E1 != E2
+            True
+            sage: E1 != E1
+            False
+        """
+        return not (self == other)
+
+    def _cmp_(self, other):
+        """
+        Arbitrary comparison function so posets work.
+        """
+        if not isinstance(other, SignedSubsetElement):
+            return 0
+        return 1
+        # x = len(self.support()) - len(other.support())
+        # if x == 0:
+        #     return x
+        # return x / abs(x) * -1
+
+    def __bool__(self):
+        r"""
+        Return whether an element is not considered a zero.
+
+        For an oriented matroid, we consider the empty set
+        `\emptyset = (\emptyset,\emptyset)` to be a zero as
+        it is the same as the all zero vector.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E1 = M.elements()[0]; E1
+            +: 0
+            -: 2
+            0: 1
+            sage: bool(E1)
+            True
+            sage: E3 = M.elements()[2]; E3
+            +:
+            -:
+            0: 0,1,2
+            sage: bool(E3)
+            False
+        """
+        if len(self.support()) > 0:
+            return True
+        return False
+
+    def __iter__(self):
+        """
+        Return an iter version of ``self``.
+        """
+        for e in self.groundset():
```

Comment:
```suggestion
        for e in self._g:
```
It is better to avoid calls to other methods in this class and use the (internal) attributes for speed (and its more natural IMO).